R/robustbayesianmetaanalysis.R

```diff
@@ -0,0 +1,2299 @@
+      prior_mu     = print(s.fit$overview[[i]]$priors[["mu"]], silent = TRUE),
+      prior_tau    = print(s.fit$overview[[i]]$priors[["tau"]], silent = TRUE),
+      prior_omega  = print(s.fit$overview[[i]]$priors[["omega"]], silent = TRUE)
+    )
+    temp_priors$addRows(temp_row)
+    
+    temp_model[["temp_priors"]] <- temp_priors
+    
+    
+    ### model information
+    temp_info <- createJaspTable(title = gettext("Information"))
+    
+    temp_info$addColumnInfo(name = "priorProb",
+                            title = gettext("P(M)"),
+                            type = "number")
+    temp_info$addColumnInfo(name = "postProb",
+                            title = gettext("P(M|data)"),
+                            type = "number")
+    temp_info$addColumnInfo(name = "marglik",
+                            title = gettext("log(MargLik)"),
+                            type = "number")
+    temp_info$addColumnInfo(name = "BF",
+                            title = BF_title,
+                            type = "number")
+    
+    temp_row <- list(
+      priorProb    = s.fit$overview[[i]]$prior_prob,
+      postProb     = s.fit$overview[[i]]$posterior_prob,
+      marglik      = s.fit$overview[[i]]$marg_lik,
+      BF           = s.fit$overview[[i]]$BF
+    )
+    temp_info$addRows(temp_row)
+    
+    temp_model[["temp_info"]] <- temp_info
+    
+    
+    ### model coeficients
+    # estimate table
+    temp_coef <- createJaspTable(title = gettext("Model Estimates"))
+    temp_coef <-
+      .RoBMA_table_fill_coef(temp_coef,
+                             s.fit$overview[[i]]$tab,
+                             s.fit$overview[[i]]$add_info,
+                             options,
+                             individual = TRUE)
+    temp_model[["temp_coef"]] <- temp_coef
+    
+    ### weights and Studies's effects
+    if (!is.null(s.fit$overview[[i]]$tab)) {
+      # weights table
+      if (any(grepl("omega", rownames(s.fit$overview[[i]]$tab)))) {
+        temp_weights <-
+          createJaspTable(title = gettextf("Estimated Weights (%s)", "\u03C9"))
+        temp_weights <-
+          .RoBMA_table_fill_weights(
+            temp_weights,
+            s.fit$overview[[i]]$tab,
+            s.fit$overview[[i]]$add_info,
+            options,
+            individual = TRUE
+          )
+        temp_model[["temp_weights"]] <- temp_weights
+      }
+      
+      # estimated studies table
+      if (any(grepl("theta", rownames(s.fit$overview[[i]]$tab)))) {
+        temp_studies <-
+          createJaspTable(title = gettextf("Estimated Studies' Effects (%s)", "\u03B8"))
+        temp_studies <-
+          .RoBMA_table_fill_studies(
+            temp_studies,
+            s.fit$overview[[i]]$tab,
+            s.fit$overview[[i]]$add_info,
+            options,
+            individual = TRUE
+          )
+        temp_model[["temp_studies"]] <- temp_studies
+        
+      }
+    }
+    
+  }
+  
+  return()
+}
+.RoBMA_plots                <-
```

Comment:
This function is a nesting hell, any chance of refactoring it into separate smaller functions?

Reply:
Never mind, that seems to be an overall style of the plotting functions... Would be nice to thing about simplifying the code, but I understand it's not a priority for now